### PR TITLE
drivers: sensor: add Innovative Sensor Technology TSic xx6 driver

### PR DESCRIPTION
**Pull request by jeppenodgaard**
Add driver for TSic 206/306/316/506F/516/716 temperature sensor.
The driver uses PWM capture driver to read a single wire with Manchester-like encoding.

### CODE REVIEW

File: drivers/sensor/tsic_xx6/Kconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2024 Vitrolife A/S
+# SPDX-License-Identifier: Apache-2.0
+
+config TSIC_XX6
+	bool "TSic xx6 driver"
+	default y
+	depends on DT_HAS_TSIC_XX6_ENABLED
```
Comment (MaureenHelm) [inline drivers/sensor/tsic_xx6/Kconfig:7]:
```suggestion
	depends on DT_HAS_IST_TSIC_XX6_ENABLED
```

File: drivers/sensor/tsic_xx6/tsic_xx6.c [hunk 2]
```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2024, Vitrolife A/S
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT tsic_xx6
```
Comment (MaureenHelm) [inline drivers/sensor/tsic_xx6/tsic_xx6.c:7]:
```suggestion
#define DT_DRV_COMPAT ist_tsic_xx6
```

File: drivers/sensor/tsic_xx6/tsic_xx6.c [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+	static struct tsic_xx6_data tsic_xx6_data_##n;                                             \
+                                                                                                   \
+	static const struct tsic_xx6_config tsic_xx6_config_##n = {                                \
+		.pwm = PWM_DT_SPEC_INST_GET(n),                                                    \
+		.lower_temperature_limit = (int8_t)DT_INST_PROP(n, lower_temperature_limit),       \
+		.higher_temperature_limit = DT_INST_PROP(n, higher_temperature_limit),             \
+		.data_bits = DT_INST_PROP(n, data_bits),                                           \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(n, &tsic_xx6_init, NULL, &tsic_xx6_data_##n, &tsic_xx6_config_##n,   \
```
Comment (MaureenHelm) [inline drivers/sensor/tsic_xx6/tsic_xx6.c:280]: `SENSOR_DEVICE_DT_INST_DEFINE`

File: dts/bindings/sensor/ist,tsic-xx6.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  TSic xx6 temperature sensor.
+  https://www.ist-ag.com/sites/default/files/downloads/ATTSic_E.pdf
+
+  Example:
+    tsic_716: tsic_716 {
+      status = "okay";
+      compatible = "tsic-xx6";
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ist,tsic-xx6.yaml:11]:
```suggestion
      compatible = "ist,tsic-xx6";
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this new sensor to `tests/drivers/build_all/sensor` to ensure it builds in CI

**Comment by jeppenodgaard** [comment]
> Please add this new sensor to `tests/drivers/build_all/sensor` to ensure it builds in CI

Thanks for the review! I've applied fixes and added a commit which should enable tsic_xx6 build to CI.

**Review by MaureenHelm — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
Needs to be rebased

**Review by teburd — DISMISSED** [review]

**Comment by jeppenodgaard** [comment]
@MaureenHelm and @teburd I rebased.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/tsic_xx6/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+zephyr_library()
```
[the remark this answers is not in the document]
Comment (jeppenodgaard) [reply drivers/sensor/tsic_xx6/CMakeLists.txt:4]: Fixed.

File: tests/drivers/build_all/sensor/pwm.dtsi [hunk 6]
```diff
@@ -0,0 +1,17 @@
+
```
[the remark this answers is not in the document]
Comment (jeppenodgaard) [reply tests/drivers/build_all/sensor/pwm.dtsi]: Thanks. Fixed.

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**